python_modules/libraries/dagster-dbt/dagster_dbt/core/resource.py

```diff
@@ -333,19 +333,21 @@ def validate_state_path(cls, state_path: Optional[str]) -> Optional[str]:
 
         return os.fspath(Path(state_path).absolute().resolve())
 
-    def _get_unique_target_path(self, *, context: Optional[OpExecutionContext]) -> Path:
+    def _get_unique_target_path(
+        self, *, context: Optional[Union[OpExecutionContext, AssetExecutionContext]]
+    ) -> Path:
         """Get a unique target path for the dbt CLI invocation.
 
         Args:
-            context (Optional[OpExecutionContext]): The execution context.
+            context (Optional[Union[OpExecutionContext, AssetExecutionContext]]): The execution context.
 
         Returns:
             str: A unique target path for the dbt CLI invocation.
         """
         unique_id = str(uuid.uuid4())[:7]
         path = unique_id
         if context:
-            path = f"{context.op.name}-{context.run.run_id[:7]}-{unique_id}"
+            path = f"{context.op_execution_context.op.name}-{context.run.run_id[:7]}-{unique_id}"
```

Comment:
`AssetExecutionContext` raises a deprecation warning if `context.op` is called, but allows you to get the underlying `op_execution_context`. `OpExecutionContext` also implements `op_execution_context`